File: Partake/app/in/partake/controller/api/debug/ErrorDBExceptionAPI.java
package in.partake.controller.api.debug;

import in.partake.base.PartakeException;
import in.partake.controller.api.AbstractPartakeAPI;
import in.partake.model.dao.DAOException;
import play.mvc.Result;

public class ErrorDBExceptionAPI extends AbstractPartakeAPI {

    public static Result get() throws DAOException, PartakeException {
        return new ErrorDBExceptionAPI().execute();
    }

    /**
     * DAOException が不意に起こった場合のテスト。
     * @return
     */
    @Override
    protected Result doExecute() throws DAOException {
        throw new DAOException();
    }

}

File: app/controllers/api/debug/ErrorDBExceptionAPI.scala
package controllers.api.debug

import controllers.api.AbstractAPI
import controllers.ActionContext
import in.partake.model.dao.DAOException
import play.api.mvc.Request
import play.api.mvc.AnyContent
import play.api.mvc.PlainResult

/**
 * DAOException が不意に起こった場合。
 */
object ErrorDBExceptionAPI extends AbstractAPI[Unit, Unit] {
  override def parseRequest(request: Request[AnyContent])(implicit context: ActionContext): Unit = {
  }

  override def executeAction(params: Unit)(implicit context: ActionContext): Unit = {
  }

  override def renderResult(values: Unit)(implicit context: ActionContext): PlainResult = {
    throw new DAOException();
  }
}
